sympy/stats/joint_rv_types.py

```diff
@@ -253,3 +248,138 @@ def NormalGamma(syms, mu, lamda, alpha, beta):
+    >>> from sympy.stats.joint_rv import marginal_distribution
+    >>> from sympy.stats.joint_rv_types import Multinomial
+    >>> from sympy import Symbol
+    >>> from sympy import symbols
+    >>> x1, x2, x3 = symbols('x1, x2, x3', nonnegative=True, integer=True)
+    >>> p1, p2, p3 = symbols('p1, p2, p3', positive=True)
+    >>> M = Multinomial('M', 3, p1, p2, p3)
+    >>> density(M)(x1, x2, x3)
+    6*p1**x1*p2**x2*p3**x3/(factorial(x1)*factorial(x2)*factorial(x3))
+    >>> marginal_distribution(M, M[0])(x1).subs(x1, 1)
+    3*p1*p2**2*p3**2/2 + 3*p1*p2**2*p3 + 3*p1*p2**2 + 3*p1*p2*p3**2 +
+    6*p1*p2*p3 + 6*p1*p2 + 3*p1*p3**2 + 6*p1*p3 + 6*p1
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_distribution
+    .. [2] http://mathworld.wolfram.com/MultinomialDistribution.html
+    """
+    if not isinstance(p[0], list):
+        p = (list(p), )
+    return multivariate_rv(MultinomialDistribution, syms, n, p[0])
+
+#-------------------------------------------------------------------------------
+# Negative Multinomial Distribution ---------------------------------------------------------
+
+class NegativeMultinomialDistribution(JointDistribution):
+
+    _argnames = ['k0', 'p']
+    is_Continuous=False
+    is_Discrete = True
+
+    def check(self, k0, p):
+        _value_check(((k0 > 0) != False) and isinstance(k0, Integer),
+                        "number of failures must be a positve integer")
+        for p_k in p:
+            _value_check((p_k >= 0) != False,
+                        "probability must be at least a positive symbol.")
+
+    @property
+    def set(self):
+        return S.Naturals0**len(self.p)
+
+    def pdf(self, *k):
+        k0, p = self.k0, self.p
+        term_1 = (gamma(k0 + sum(k))*(1 - sum(p))**k0)/gamma(k0)
+        term_2 = Mul.fromiter([pi**ki/factorial(ki) for pi, ki in zip(p, k)])
+        return term_1*term_2
+
+def NegativeMultinomial(syms, k0, *p):
+    """
+    Creates a discrete random variable with Negative Multinomial Distribution.
+
+    The density of the said distribution can be found at [1].
+
+    Parameters
+    ==========
+    k0: postive integer of class Integer,
+        number of failures before the experiment is stopped
+    p: event probabilites, >= 0 and <= 1
+
+    Returns
+    =======
+    A RandomSymbol.
+
+    Examples
+    ========
+    >>> from sympy.stats import density
+    >>> from sympy.stats.joint_rv import marginal_distribution
+    >>> from sympy.stats.joint_rv_types import NegativeMultinomial
+    >>> from sympy import symbols
+    >>> x1, x2, x3 = symbols('x1, x2, x3', nonnegative=True, integer=True)
+    >>> p1, p2, p3 = symbols('p1, p2, p3', positive=True)
+    >>> N = NegativeMultinomial('M', 3, p1, p2, p3)
+    >>> N_c = NegativeMultinomial('M', 3, 0.1, 0.1, 0.1)
+    >>> density(N)(x1, x2, x3)
+    p1**x1*p2**x2*p3**x3*(-p1 - p2 - p3 + 1)**3*gamma(x1 + x2 + x3 + 3)/
+    (2*factorial(x1)*factorial(x2)*factorial(x3))
```

Comment:
@sidhantnagpal  Lines 372 and 373 when written on a single line are getting too long. [Doctests](https://travis-ci.org/sympy/sympy/jobs/531121583) are failing due to writing the result in two lines. Is there something like `\` for docstrings?

Reply:
Would it be possible to make use of `#doctest: +NORMALIZE_WHITESPACE`?

Reply:
I exploited the property of  continuation in `()`, like,
```
>>> some function
(a + 
b)
```
The above worked well. Though found [this](https://docs.python.org/2/library/doctest.html#how-are-docstring-examples-recognized). Thanks for the comment.